forum/utils.py

```diff
@@ -30,3 +33,74 @@ def handle_proxy_requests(request, suffix, method):
         headers=request_headers,
         timeout=5.0,
     )
+
+
+def flag_as_abuse(user, entity):
+    """
+    Flag an entity as abuse.
+
+    Args:
+        user (dict): The user who is flagging the entity as abuse.
+        entity (dict): The entity being flagged as abuse.
+
+    Returns:
+        dict: The updated entity with the abuse flag.
+
+    Raises:
+        ValueError: If user ID or entity is not provided.
+    """
+    if not (user and entity):
+        raise ValueError("User ID/Entity is required")
+    # Check if user ID is already in abuse flaggers list
+    abuse_flaggers = entity["abuse_flaggers"]
+    if user["_id"] not in abuse_flaggers:
+        abuse_flaggers.append(user["_id"])
+        Contents().collection.update_one(
+            {"_id": entity["_id"]},
+            {"$set": {"abuse_flaggers": abuse_flaggers}},
+        )
+
+    # Check if this is the first abuse flag
+    first_flag_added = len(entity["abuse_flaggers"]) == 1
+
+    # If this is the first abuse flag, update author's stats
+    if first_flag_added:
+        Users().collection.update_one(
+            {"_id": entity["author_id"]},
+            {"$inc": {"active_flags": 1}},
+        )
+
+    # Reload the object and return it as a JSON string
+    return Contents().get(_id=entity["_id"])
+
+
+def un_flag_as_abuse(user, entity, _all=True):
+    """
+    Unflag an entity as abuse.
+
+    Args:
+        user (dict): The user who is unflagging the entity as abuse.
+        entity (dict): The entity being unflagged as abuse.
+        all (bool, optional): Whether to remove all abuse flags or just the current user's flag. Defaults to True.
+
+    Returns:
+        dict: The updated entity with the abuse flag removed.
+
+    Raises:
+        ValueError: If user ID or entity is not provided.
+    """
+    if not (user and entity):
+        raise ValueError("User ID/Entity is required")
+    if _all:
+        Contents().collection.update_one(
```

Comment:
Let's put this in a separate function, e.g: `unflag_all_as_abuse`. Boolean arguments are a red flag: if a function should behave one way or another based on a flag value, then maybe we should just split the function in two parts.

Reply:
Done